### PR TITLE
update lightning integration example

### PR DESCRIPTION
**Pull request by rohitgr7**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

Updated the lightning integration example to put some more light on when `.reset` will be called automatically.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/pages/lightning.rst [hunk 1]
```diff
@@ -32,12 +32,13 @@ The example below shows how to use a metric in your `LightningModule <https://py
             preds = self(x)
             ...
             # log step metric
-            self.log('train_acc_step', self.accuracy(preds, y))
```
Comment (rohitgr7) [inline docs/source/pages/lightning.rst:35]: since `self.accuracy(preds, y)` returns a tensor, the metric instance won't reset automatically because lightning doesn't have access to the instance.

File: docs/source/pages/lightning.rst [hunk 2]
```diff
@@ -32,12 +32,13 @@ The example below shows how to use a metric in your `LightningModule <https://py
...
+            self.accuracy(preds, y)
+            self.log('train_acc_step', self.accuracy)
             ...
 
         def training_epoch_end(self, outs):
             # log epoch metric
-            self.log('train_acc_epoch', self.accuracy.compute())
```
Comment (rohitgr7) [inline docs/source/pages/lightning.rst:40]: same here.

**Comment by rohitgr7** [comment]
tests are failing on master

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: docs/source/pages/lightning.rst [hunk 3]
```diff
@@ -86,6 +87,43 @@ If ``on_epoch`` is True, the logger automatically logs the end of epoch metric v
             self.valid_acc(logits, y)
             self.log('valid_acc', self.valid_acc, on_step=True, on_epoch=True)
 
+.. note:: the ``.reset()`` method of the metric will automatically be called at the end of an epoch within lightning only if you pass
+    the metric instance inside `self.log <https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#logging-from-a-lightningmodule>`_.
+    Also if you are calling ``.compute`` by yourself, you need to call the ``.reset()`` too.
```
[the remark this answers is not in the document]
Comment (SkafteNicki) [reply docs/source/pages/lightning.rst:92]: The sentence refer to the case where the user calls `compute` themself, and in that case they need to call `reset` themself
